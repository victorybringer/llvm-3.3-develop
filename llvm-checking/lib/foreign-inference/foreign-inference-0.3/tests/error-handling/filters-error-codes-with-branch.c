#include <unistd.h>

#define EOF 1
#define OK 0
#define FATAL (-1)

int errGen(int fd) {
  char buffer[100];
  int bs;

  bs = read(fd, buffer, 10);
  if(bs < 0)
    return FATAL;

  bs = write(fd, buffer, 1);
  if(bs < 0)
    return EOF;

  return OK;
}

int target(int fd) {
  int rc = errGen(fd);

  if(rc == EOF)
    return FATAL;

  if(rc != OK)
    return rc;

  return 5;
}
