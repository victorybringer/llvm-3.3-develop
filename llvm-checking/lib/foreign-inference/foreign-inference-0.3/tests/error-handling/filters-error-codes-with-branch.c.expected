fromList [ ("target", (fromList [], ReturnConstantInt (fromList [-1])))
         , ("errGen", (fromList [], ReturnConstantInt (fromList [-1,1])))
         ]